### PR TITLE
Add Java web scraper to get small business names.

### PR DESCRIPTION
**Pull request by gannerr**
Add Java web scraper to get small business names.

### CODE REVIEW

File: capstone/pom.xml [hunk 1]
```diff
@@ -23,6 +23,31 @@
...
+      <groupId>com.google.appengine</groupId>
+      <artifactId>appengine-api-1.0-sdk</artifactId>
+      <version>1.9.59</version>
+    </dependency>
+    <dependency>
+      <groupId>com.google.appengine</groupId>
+      <artifactId>appengine-api-1.0-sdk</artifactId>
+      <version>1.9.59</version>
+    </dependency>
+    <dependency>
```
Comment (graceehuu) [inline capstone/pom.xml:41]: seems to be duplicates?

File: capstone/src/main/java/com/google/sps/servlets/scraperServlet.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      String restaurantNameSelector = "div > h4 > a";
+      Elements restaurantNameElements = page.select(restaurantNameSelector);
+
+
+      for (Element restaurantName : restaurantNameElements) {
+        restaurantNames.add(restaurantName.text());
+      }
+    }
+    
+    System.out.println(restaurantNames);
```
Comment (graceehuu) [inline capstone/src/main/java/com/google/sps/servlets/scraperServlet.java:52]: remove debugging

File: capstone/src/main/java/com/google/sps/servlets/scraperServlet.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+
+/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+@WebServlet("/gimmeSmallBizNames")
```
Comment (graceehuu) [inline capstone/src/main/java/com/google/sps/servlets/scraperServlet.java:30]: lol nice

File: capstone/src/main/java/com/google/sps/servlets/scraperServlet.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    String urlbase = "https://www.helpourneighborhoodrestaurants.com/";
+    String[] locations = {"brooklyn", "manhattan", "queens", "staten-island", "bronx"};
+
+    for (String location : locations) {
+      String url = urlbase.concat(location);
+      Document page = Jsoup.connect(url).userAgent("JSoup Scraper").get();
+
+      String restaurantNameSelector = "div > h4 > a";
+      Elements restaurantNameElements = page.select(restaurantNameSelector);
+
```
Comment (graceehuu) [inline capstone/src/main/java/com/google/sps/servlets/scraperServlet.java:45]: nit: extra line

File: capstone/pom.xml (hunk 1 above)
Comment (gannerr) [reply capstone/pom.xml:41]: fax

File: capstone/src/main/java/com/google/sps/servlets/scraperServlet.java (hunk 2 above)
Comment (gannerr) [reply capstone/src/main/java/com/google/sps/servlets/scraperServlet.java:52]: mb

File: capstone/src/main/webapp/script.js [hunk 5]
```diff
@@ -159,3 +159,13 @@ function displayPanel(name) {
   document.getElementById("panel").style.display = "block";
   document.getElementById("restaurant-info").innerHTML = name;
 }
+
+function getSmallBizNames() {
+  fetch('/gimmeSmallBizNames').then(response => response.json()).then((restaurantNames) => {
```
Comment (terencedignon) [inline capstone/src/main/webapp/script.js:164]: lol I don't want to crush your creativity but I think it's better to make the fetching as neutral as possible and to use common lingo. In google if you are making an API call we use 'fetch' as a prefix. So in this case `fetchPlaceNames` or `fetchBusinessNames` would be more appropriate.

File: capstone/target/maven-status/maven-compiler-plugin/compile/default-compile/inputFiles.lst [hunk 6]
```diff
@@ -1 +1,2 @@
-/home/casanted/step196-2020/capstone/src/main/java/com/google/sps/servlets/DataServlet.java
+/home/andrewgan/step196-2020/capstone/src/main/java/com/google/sps/servlets/DataServlet.java
```
Comment (terencedignon) [inline capstone/target/maven-status/maven-compiler-plugin/compile/default-compile/inputFiles.lst:1]: This seems like it should be gitignored?

File: capstone/target/portfolio-1/script.js [hunk 7]
```diff
@@ -159,3 +159,13 @@ function displayPanel(name) {
   document.getElementById("panel").style.display = "block";
   document.getElementById("restaurant-info").innerHTML = name;
 }
+
+function getSmallBizNames() {
+  fetch('/gimmeSmallBizNames').then(response => response.json()).then((restaurantNames) => {
+    for(let name of restaurantNames) {
+      console.log(name);
```
Comment (terencedignon) [inline capstone/target/portfolio-1/script.js:166]: Why are we committing the console log? I'm okay with it as long as there's a reason it's there to help further development. That said, there should be an issue attached for the plan to remove it.

File: capstone/src/main/webapp/script.js (hunk 5 above)
Comment (gannerr) [reply capstone/src/main/webapp/script.js:164]: fair enough

File: capstone/target/portfolio-1/script.js (hunk 7 above)
Comment (gannerr) [reply capstone/target/portfolio-1/script.js:166]: just wanted to make sure i was getting good outputs. yeah gonna put an issue to remove it and replace with insert names into hashset.

**Pull request merged**